### PR TITLE
DomainSocket support

### PR DESCRIPTION
**Pull request by czp3009**
This PR aims to implement support for DomainSockets, using code from https://github.com/jasync-sql/jasync-sql/pull/176.

But the test code still has some problems. Since the project's dependencies do not include "netty-transport-native-epoll", the `Epoll` class does not exist even when test on Linux system. Thus, `NettyUtils` always chose `NioSocketChannel`.

But adding "netty-transport-native-epoll" to the dependency will cause some tests for `NettyUtils` to fail because they are hard-coded and the test code will always determine if `NioSocketChannel` is returned.

I'm confused about how to test all the code, both without the domain socket and with it. Further, in order to test domain sockets, then all test code may not work on some systems, such as Windows.

### CODE REVIEW

File: mysql-async/src/main/java/com/github/jasync/sql/db/mysql/codec/MySQLConnectionHandler.kt [hunk 1]
```diff
@@ -4,54 +4,19 @@ import com.github.jasync.sql.db.Configuration
...
-import com.github.jasync.sql.db.mysql.message.server.BinaryRowMessage
-import com.github.jasync.sql.db.mysql.message.server.ColumnDefinitionMessage
-import com.github.jasync.sql.db.mysql.message.server.EOFMessage
-import com.github.jasync.sql.db.mysql.message.server.ErrorMessage
-import com.github.jasync.sql.db.mysql.message.server.HandshakeMessage
-import com.github.jasync.sql.db.mysql.message.server.OkMessage
-import com.github.jasync.sql.db.mysql.message.server.PreparedStatementPrepareResponse
-import com.github.jasync.sql.db.mysql.message.server.ResultSetRowMessage
-import com.github.jasync.sql.db.mysql.message.server.ServerMessage
+import com.github.jasync.sql.db.mysql.message.client.*
```
Comment (oshai) [inline mysql-async/src/main/java/com/github/jasync/sql/db/mysql/codec/MySQLConnectionHandler.kt:7]: please refrain from using star imports

Comment (czp3009) [reply mysql-async/src/main/java/com/github/jasync/sql/db/mysql/codec/MySQLConnectionHandler.kt:7]: My IDEA with default JetBrains code style automatic format to *, could you please provide your code style xml so i can import to my IDEA.

**Comment by czp3009** [comment]
> I think a reasonable approach can be to control that with system properties like "forceNioSocket" or "forceDomainSocket" that will be set by specific tests.

hi @oshai , I'm sorry, but i couldn't find the description of system properties in the netty document, cloud you please provide the documentation page about that so I can learn it. Thanks!

**Comment by oshai** [comment]
> > I think a reasonable approach can be to control that with system properties like "forceNioSocket" or "forceDomainSocket" that will be set by specific tests.
> 
> hi @oshai , I'm sorry, but i couldn't find the description of system properties in the netty document, cloud you please provide the documentation page about that so I can learn it. Thanks!

I was referring java system properties: https://docs.oracle.com/javase/tutorial/essential/environment/sysprop.html

File: mysql-async/src/main/java/com/github/jasync/sql/db/mysql/codec/MySQLConnectionHandler.kt (hunk 1 above)
Comment (oshai) [reply mysql-async/src/main/java/com/github/jasync/sql/db/mysql/codec/MySQLConnectionHandler.kt:7]: see https://stackoverflow.com/questions/3348816/intellij-never-use-wildcard-imports/25916603

Comment (czp3009) [reply mysql-async/src/main/java/com/github/jasync/sql/db/mysql/codec/MySQLConnectionHandler.kt:7]: Thanks

**Comment by czp3009** [comment]
> > > I think a reasonable approach can be to control that with system properties like "forceNioSocket" or "forceDomainSocket" that will be set by specific tests.
> > 
> > 
> > hi @oshai , I'm sorry, but i couldn't find the description of system properties in the netty document, cloud you please provide the documentation page about that so I can learn it. Thanks!
> 
> I was referring java system properties: https://docs.oracle.com/javase/tutorial/essential/environment/sysprop.html

There may be some misunderstanding, I am not asking how to read and set system properties. I want to know what "forceNioSocket" or "forceDomainSocket" does, and what documentation they are described in.

Obviously, they are not java standard system properties.

**Comment by oshai** [comment]
> > > > I think a reasonable approach can be to control that with system properties like "forceNioSocket" or "forceDomainSocket" that will be set by specific tests.
> > > 
> > > 
> > > hi @oshai , I'm sorry, but i couldn't find the description of system properties in the netty document, cloud you please provide the documentation page about that so I can learn it. Thanks!
> > 
> > 
> > I was referring java system properties: https://docs.oracle.com/javase/tutorial/essential/environment/sysprop.html
> 
> There may be some misunderstanding, I am not asking how to read and set system properties. I want to know what "forceNioSocket" or "forceDomainSocket" does, and what documentation they are described in.
> 
> Obviously, they are not java standard system properties.

What I meant is that in `NettyUtils` we will have code to check for system properties and in that case, use only the "forced" class without checking other types for existence.

**Comment by czp3009** [comment]
> > > > > I think a reasonable approach can be to control that with system properties like "forceNioSocket" or "forceDomainSocket" that will be set by specific tests.
> > > > 
> > > > 
> > > > hi @oshai , I'm sorry, but i couldn't find the description of system properties in the netty document, cloud you please provide the documentation page about that so I can learn it. Thanks!
> > > 
> > > 
> > > I was referring java system properties: https://docs.oracle.com/javase/tutorial/essential/environment/sysprop.html
> > 
> > 
> > There may be some misunderstanding, I am not asking how to read and set system properties. I want to know what "forceNioSocket" or "forceDomainSocket" does, and what documentation they are described in.
> > Obviously, they are not java standard system properties.
> 
> What I meant is that in `NettyUtils` we will have code to check for system properties and in that case, use only the "forced" class without checking other types for existence.

Got it!

**Comment by oshai** [comment]
@czp3009 any update on this PR?

**Comment by czp3009** [comment]
> @czp3009 any update on this PR?

I'm sorry, I'm still trying to update this PR, but I've been very busy at work and haven't been able to find the time. If I have time I will add the previously missing test code to this PR immediately

**Comment by oshai** [comment]
> Add support for unix domain socket #175

Thanks! note that I continue work on that in #319.

**Comment by czp3009** [comment]
Thank you very much for your awesome work, since unix domain socket has been implemented, I will close the PR

**Pull request closed without merging**